query-service-impl/src/main/java/org/hypertrace/core/query/service/projection/ProjectionTransformation.java

```diff
@@ -159,17 +174,49 @@
       case INT_VALUE:
         return Single.just(createLongLiteralExpression(literal.getIntValue()));
       case VALUE_NOT_SET:
+        return this.getCorrespondingNullExpression(expectedType);
       default:
         return Single.error(
             new UnsupportedOperationException("Unrecognized literal type: " + literal));
     }
   }
 
+  private Single<Expression> getCorrespondingNullExpression(AttributeKind attributeKind) {
+    switch (attributeKind) {
+      case TYPE_DOUBLE:
+      case TYPE_DOUBLE_ARRAY:
+      case TYPE_INT64:
+      case TYPE_INT64_ARRAY:
+      case TYPE_TIMESTAMP:
+        return Single.just(createNullNumberLiteralExpression());
+      case TYPE_BOOL:
+      case TYPE_BOOL_ARRAY:
+      case TYPE_STRING:
+      case TYPE_STRING_ARRAY:
+      case TYPE_STRING_MAP:
+      case TYPE_BYTES:
+        return Single.just(createNullStringLiteralExpression());
+      case UNRECOGNIZED:
+      case KIND_UNDEFINED:
+      default:
+        return Single.error(
+            new UnsupportedOperationException(
+                "Unrecognized attribute kind for null expression conversion: " + attributeKind));
+    }
+  }
+
   private Single<Expression> rewriteProjectionExpressionAsQueryExpression(
       ProjectionExpression projectionExpression) {
+    List<Projection> arguments = projectionExpression.getArgumentsList();
+    // TODO fallback for unknown projections
```

Comment:
fixing this one?

Reply:
good thing I put a todo in, forgot to come back to it :) will update.

Reply:
good thing I had taken off my trust cap :)

Reply:
Updated - I wrote a test for it, but it's actually not testable since we cover all known projections in code and its an enum. I modified the code to not cover one case and ran the test :) Any other ideas for testing the situation where it's an unknown enum value?

Reply:
Nope. No idea lol. Was gonna say you could pass in a int value but compiler will not let you. And if you some reflection magic, then the reflection fails when looking for the enum to convert the int to.